app/src/userpages/components/SnippetDialog/index.jsx

```diff
@@ -1,54 +1,26 @@
-// @flow
-
-import React, { useState, useCallback, useEffect, useMemo } from 'react'
+import React, { useState, useCallback, useRef } from 'react'
 import { I18n, Translate } from 'react-redux-i18n'
-
 import ModalPortal from '$shared/components/ModalPortal'
 import Dialog from '$shared/components/Dialog'
 import Buttons from '$shared/components/Buttons'
 import DropdownActions from '$shared/components/DropdownActions'
-import CodeSnippet from '$shared/components/CodeSnippet'
+import { CodeSnippet, titleize } from '@streamr/streamr-layout'
 
-import { ProgrammingLanguages, StreamrClientRepositories } from '$shared/utils/constants'
+import { StreamrClientRepositories } from '$shared/utils/constants'
 import useCopy from '$shared/hooks/useCopy'
 
 import styles from './snippetDialog.pcss'
 
-type Language = $Values<typeof ProgrammingLanguages>
-
-type Props = {
-    snippets: {
-        [Language]: string,
-    },
-    onClose: () => void,
-}
-
-const SnippetLanguageMappings = {
-    [ProgrammingLanguages.JAVASCRIPT]: 'javascript',
-    [ProgrammingLanguages.JAVA]: 'java',
-}
-
-const SnippetDialog = ({ snippets, onClose }: Props) => {
+const SnippetDialog = ({ snippets, onClose }) => {
     const { isCopied, copy } = useCopy()
-    const [selectedLanguage, setSelectedLanguage] = useState(undefined)
-
-    const onSelectLanguage = useCallback((language: Language) => {
-        setSelectedLanguage(language)
-    }, [])
 
-    const snippetLanguages = useMemo(() => Object.keys(snippets), [snippets])
+    const [selectedLanguage, setSelectedLanguage] = useState(Object.keys(snippets)[0])
 
-    useEffect(() => {
-        if (!selectedLanguage) {
-            setSelectedLanguage(snippetLanguages.length > 0 && snippetLanguages[0])
-        }
-    }, [selectedLanguage, snippetLanguages])
+    const codeRef = useRef({})
```

Comment:
`codeRef` holds nicely formatted (reindented) snippets rendered by `CodeSnippet` component. `codeRef.current[lang]` can be used right away for copying, etc.

Reply:
Maybe even the `language` property is unnecessary... could just hold the currently displayed code?

Reply:
We'd have to keep track on which one is active. I'll pass.

Reply:
Sure, it works this way as well... But isn't this effectively the same as doing this in the component if the ref isn't kept in the `CodeSnippet `:

```js
const codeRef = useRef('')

useEffect(() => {
     codeRef.current = snippets[selectedLanguage]
}, [snippets, selectedLanguage])

const onCopy = useCallback(() => {
     copy(codeRef.current || '')
}, [copy])

...

<CodeSnippet showLineNumbers>
     {codeRef.current}
</CodeSnippet>
```

Reply:
I see your point. `snippets` (generated using `streamSnippets`) is a collection of badly indented codes. `CodeSnippet` does the reindenting internally and stores the final codes in the "registery" (given ref). Does it make sense?

Btw, your above code would work with `useState` but won't with `useRef`, especially the rendering part (`ref.current=` doesn't trigger a rerender).

Reply:
Ok, fair enough - let's keep it like it is 👍